scapy/arch/solaris.py

```diff
@@ -6,11 +6,1576 @@
 """
 Customization for the Solaris operation system.
 """
+from __future__ import with_statement
+import sys,os,struct,socket,time,platform
+from select import select
+import fcntl
+from fcntl import ioctl
+import scapy
+import scapy.utils
+import scapy.utils6
+from scapy.config import conf
+from scapy.supersocket import SuperSocket
+import scapy.arch
+from scapy.error import warning,Scapy_Exception,log_loading
+import scapy.data
+import scapy.plist as plist
+import scapy.sendrecv
+from scapy.utils import *
 
-# IPPROTO_GRE is missing on Solaris
-import socket
-socket.IPPROTO_GRE = 47
+
+if "sunos" in sys.platform:
+    try:
+        import dlpi
+        if not hasattr(dlpi, "DL_PROMISC_NOLOOP"):
+            dlpi.DL_PROMISC_NOLOOP = 0x10000000
+        conf.use_dlpi = 1
+    except ImportError,e:
+        log_loading.info("Solaris Unable to import dlpi module: %s" % e)
+        conf.use_dlpi = 0
+else:
+    raise Exception("system platform need be solaris")
+
+log_loading.warning("Solaris must indicate a speicific network interface when recieve msg")
+log_loading.warning("It means sr()/sniff() need 'iface' parameter.")
+log_loading.warning("Otherwise, the interface which has default route will be used.")
+
+X86_64 = ( platform.architecture()[0] == '64bit')
+
+
+# correct data.py value
+scapy.data.ETH_P_ALL = 0
+scapy.sendrecv.ETH_P_ALL = 0
+from scapy.data import *
+
+def load_ethertypes(filename):
```

Comment:
Which line?
1.  line 43, os.uname()[4] at Solaris's PYTHON always return 'i86pc' or 'sun4v', so I have to use platform.architecture()[0] to judge if it is 64bit.
2. line 47,  scapy.data.ETH_P_ALL is defined as 0 at Solaris
3. line 51, load_ethertypes, Solaris has no '/etc/ethertypes', so I have to provide raw input.

I can't find a way to make Solaris bound to 'any' interafces like Linux, whatever AF_PACKET or libpcap, so I have to limit the function of sr()/sniff() that it must be bound to an interface.